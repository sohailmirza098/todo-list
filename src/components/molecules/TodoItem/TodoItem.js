import React from 'react'
import Checkbox from '../../atoms/CheckboxInput/Checkbox';
import RemoveButton from '../../atoms/RemoveButton/RemoveButton';


export default function TodoItem(props) {
  
  const getStyle = () => {
    console.log(props.todo);
    return {
      background: '#f4f4f4',
      padding: '10px',
      borderBottom: '1px #ccc dotted',
      textDecoration: props.todo.completed ? 'line-through' : 'none'
    }
  }

  // const {title} = props.todo;

  return (
    
    <div style={getStyle()} data-testid="todoid" >
        <Checkbox markComplete={ props.markComplete } id={props.todo.id} />
        { props.todo.title }
        <RemoveButton id={props.todo.id} delTodo={ props.delTodo } value={'x'}/>

    </div>
  )
}
