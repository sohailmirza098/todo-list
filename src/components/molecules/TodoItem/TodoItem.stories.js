import React from 'react';
import TodoItem from "./TodoItem";
import { action } from '@storybook/addon-actions';
// import { text } from '@storybook/addon-knobs/react';
import { ulid } from 'ulid';

export default {
  title: 'TodoItem',
  component: TodoItem,
};

let props = {
    "data-testid": "todoid",
    key: ulid(),
    todo: {
        id: 1,
        title: "title",
        completed: false
    }
}

export const input = () => {
    return(<TodoItem key={ props.key } todo={props.todo} />);
};
