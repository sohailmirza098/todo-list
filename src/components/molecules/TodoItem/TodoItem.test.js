import React from "react";
import { create } from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import TodoItem from './TodoItem';
import { ulid } from 'ulid';

describe("Todos testing", () => {
    let props = {
        "data-testid": "todoid",
        key: ulid(),
        todo: {
            id: 1,
            title: "title",
            completed: false
        },
        markComplete: jest.fn(),
        delTodo: jest.fn()
    }

  test("Matches the snapshot", () => {
    const input = create(<TodoItem  key={ props.key } todo={props.todo} />);
    expect(input.toJSON()).toMatchSnapshot();
  });

  test("TodoItem is present", () => {
    const {getByTestId}=  render(<TodoItem key={ props.key } todo={props.todo}/>);
    const header = getByTestId("todoid");
    expect(header).toBeInTheDocument();
  });

});