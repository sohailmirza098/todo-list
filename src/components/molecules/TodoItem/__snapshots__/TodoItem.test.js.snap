exports[`Todos testing Matches the snapshot 1`] = `
<div
  data-testid="todoid"
  style={
    Object {
      "background": "#f4f4f4",
      "borderBottom": "1px #ccc dotted",
      "padding": "10px",
      "textDecoration": "none",
    }
  }>
  <input
    onChange={[Function]}
    type="checkbox" />
  title
  <button
    onClick={[Function]}
    style={
      Object {
        "background": "#ff0000",
        "border": "none",
        "borderRadius": "50%",
        "color": "#fff",
        "cursor": "pointer",
        "float": "right",
        "padding": "5px 9px",
      }
    }>
    x
  </button>
</div>
`;
